routing/base/astar_algorithm.hpp

```diff
@@ -204,7 +220,8 @@ typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPath(
 
     if (stateV.vertex == finalVertex)
     {
-      ReconstructPath(stateV.vertex, parent, path);
+      ReconstructPath(stateV.vertex, parent, result.path);
+      result.distance = stateV.distance;
```

Comment:
stateV.distance is a reduced distance, please, do not use it outside of the A* algorithm.